File: java/src/test/PublishTest.java
package mx.iteso;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
import static org.junit.Assert.Equals;
import static org.junit.Assert.assertTrue;
import static org.junit.Before;
import static org.junit.Test;

/**
 * Publish test class.
 */
public class PublishTestClass {
    private User user;

    /**
     * Set up method.
     */
    @Before
    public void SetUp(){
        user = mock((User.class));
    }    
    /**
     * Get privilege test.
     */
    @Test
    public void PrivilegeTest(){
        User publish = new Approve(user);
        when(user.getPrivileges()).thenReturn("Any Name publish");
    }
}

File: csharp/decorator_practice/decorator_practice/test/PublishTest.cs
using System;
using System.Collections.Generic;
using System.Text;
using decorator_practice.main.decorator;
using decorator_practice.main.privileges;
using decorator_practice.main.users;
using Moq;
using NUnit.Framework;

namespace decorator_practice.test
{
    [TestFixture]
    class PublishTest
    {
        public Mock<User> user;

        [SetUp]
        public void SetUp()
        {
            user = new Mock<User>();
        }

        [TestCase]
        public void NameTest()
        {
            User publish = new Publish(user.Object);
            Assert.AreEqual("Any Name publish", user.Object.getPrivileges());
        }
    }
}